GOVERNANCE.md

```diff
@@ -0,0 +1,477 @@
+
+  Commits using the `--fixup` or `--squash` feature of `git-commit` are allowed,
+  but won't be merged (read more about this in the "merge strategies" section).
+
+- **Ensuring the Signed-off-by-trailer**
+  Contributors need to sign the "Contributor License Agreement". They do so by
+  signing off each individual commit with a so-called "Signed-off-by Trailer".
+  Git offers the `--signoff`/`-s` flags when using `git-commit` to commit
+  changes.
+
+  To ensure that all commits have the trailer present in the commit message, a
+  CI lint is installed in the github actions workflows of the project. This lint
+  blocks pull requests to be merged if one or more of the commits of the pull
+  request misses the `Signed-off-by` trailer.
+
+  As a result, it is not possible to merge pull requests that miss the trailer.
+- **Coding styleguide**
+  Coding style is enforced via `rustfmt` in its default configuration.
+  Compliance with the coding style is enforced via CI.
+- **Testing**
+  Testing is done via workflows in github actions using `cargo test --all-features` for all
+  crates.
+  The main objective with this is that a developer should be able to simply run
+  `cargo test` on their local machine to be able to see whether CI would succeed
+  for changes they submit in a pull request.
+  Thus, all CI tests (unit tests, but also integration tests) are implemented in
+  Rust and do not rely on external services.
+  End-to-end system tests - that depend on external services - are run outside the CI pipeline,
+  to avoid inconsistent test outcomes because of external issues.
+- **Benchmarks**
+- **Documentation builds**
+  Source code documentation as well as other documentation is tested via github
+  actions workflows as well, to ensure that a developer is able to build all
+  relevant documentation on their local machine.
+- **Keeping spec up to date**
+- **Evergreen master**
+  The project pursues the "evergreen master" strategy. That means that at every
+  point in time, the commit that `master`/`main` points to must successfully
+  build and pass all tests successfully.
+
+  Reaching that goal is possible because of the employed merge strategies (read
+  below).
+- **Merge strategies**
+  Merging is the way how the project accepts and implements changes.
+
+  Because of the pursued "everygreen master", merging is implemented in a
+  certain way that forbids several bad practices which could lead to a breaking
+  build on the `master`/`main` branch, and a special workflow is implemented to
+  ensure not only the "evergreen master" but also to streamline the workflow for
+  integrating pull requests.
+
+  The following actions are **not** allowed:
+
+  * Committing to `master`/`main` directly
+  * Squash-Merging of pull requests (which is equivalent to committing to
+    `master`/`main` directly).
+    The github repository is configured to not allow squash merges.
+  * Merging of "!fixup" or "!squash" commits. A github actions
```

Comment:
```suggestion
  * Merging of "fixup!" or "squash!" commits. A github actions
```

It is actually called "fixup!" instead of "!fixup". I know this is a nitpick... :wink:

Reply:
Fixed